### PR TITLE
Feature: Yank buffer to system clipboard

### LINKED ISSUE #6: [Feature] Yank to System Clipboard
This issue serves as a record of a proposed feature for this extension. I'll be implementing it myself soon (probably tonight, if I have nothing better to do), but having a place for discussion might be helpful.

## Rationale

Since YAGPDB CCs are managed though the online dashboard, they must be copy/pasted when edits are made in another platform. With a standard GUI editor like VSCode, this isn't too painful, especially with the <kbd>Ctrl + A</kbd> shortcut, but when using a terminal editor like Vim, it can be painful (especially if you have line numbers enabled, or another pane on the side). As such, a function (which can be bound to a shortcut by the user) would be quite useful.

## Notes

(N)Vim can access the system clipboard using the `"*` and `"+` registers. `"+` *seems* to be the preferred one, though I'm not certain about that. On Windows, both appear to work equivalently, and do in fact add to the Clipboard History feature.

Syntax here is just like with any other register - yank to it using `"+y{motion}` or `{Visual}"+y`, and paste with `"+p`.

### PR DESCRIPTION
**Pull request by LRitzdorf**
**Please describe the changes this pull request does and why it should be merged:**

Adds a command to quickly copy the entire buffer to the system clipboard, via the `"*` register.

Further details discussed in #6. Closes #6.

**Terms**
- [X] I agree to follow this project's [Code of Conduct](CODE_OF_CONDUCT.md)
- [X] I have read and understood this project's [Contributing Guidelines](../CONTRIBUTING.md)

### CODE REVIEW

**Review by l-zeuch — COMMENTED** [review]
Does wrapping the yank to clipboard also echo `Yanked x lines to "+`?
If not, perhaps add a feedback that it did exactly that, might be confusing otherwise..

**Comment by LRitzdorf** [comment]
It does indeed. The command used is `execute '%y *'`, so it behaves just as if the user had typed that themselves (including the "Yanked ... lines" response).

File: ftplugin/yagpdbcc/functions.vim [hunk 1]
```diff
@@ -43,11 +43,21 @@ function! s:NextSection(type, backwards, visual)
...
 vnoremap <script> <buffer> <silent> [[ :<c-u>call <SID>NextSection(1, 1, 1)<cr>
 vnoremap <script> <buffer> <silent> ][ :<c-u>call <SID>NextSection(2, 0, 1)<cr>
 vnoremap <script> <buffer> <silent> [] :<c-u>call <SID>NextSection(2, 1, 1)<cr>
+
+" Quick function and command to copy the whole file to the system clipboard
+function! YagCopy()
+    if has('clipboard')
+        execute '%y *'
+    else
+        echo "Your vim doesn't appear to have clipboard support."
```
Comment (l-zeuch) [inline ftplugin/yagpdbcc/functions.vim:60]:
```suggestion
        if has('nvim')
            echo "Your neovim doesn't appear to have clipboard support."
        else
            echo "Your vim doesn't appear to have clipboard support."
        endif
```

**Review by l-zeuch — CHANGES_REQUESTED** [review]
This one might be super nitpicky...

File: ftplugin/yagpdbcc/functions.vim (hunk 1 above)
Comment (LRitzdorf) [reply ftplugin/yagpdbcc/functions.vim:60]: After further research, Neovim is always compiled with clipboard support. (Of course, you can build it yourself, but in that case you're smart enough not to need a custom help message like this one :) ). Closing.

Comment (l-zeuch) [reply ftplugin/yagpdbcc/functions.vim:60]: Of course you beat me to it :)

**Review by l-zeuch — APPROVED** [review]

**Pull request merged**